packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1600,6 +1600,25 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
 
   bool _debugLocked = false; // used to prevent re-entrant calls to push, pop, and friends
 
+  bool _shouldAbandonInitialRoute(List<Route<dynamic>> plannedInitialRoutes) {
+    assert(plannedInitialRoutes.isNotEmpty);
+
+    // The last route has to match something in order for it to work.
+    if (plannedInitialRoutes.last == null) {
+      return true;
+    }
+
+    // In web, we don't care if there are gaps in the initial route.
```

Comment:
Why this commented out code?

Reply:
Removed. The commented out code was supposed to make the change only affect the web. But since I'm making this change for all platforms now, I removed the method entirely.